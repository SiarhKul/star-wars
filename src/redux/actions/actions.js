const SET_PEOPLE_TO_STORE = "SET_PEOPLE_TO_STORE";
const SET_PLANETS_TO_STORE = "SET_PLANETS_TO_STORE";
const SET_STARSHIPS_TO_STORE = "SET_STARSHIPS_TO_STORE";
const SET_FILMS_TO_STORE = "SET_FILMS_TO_STORE";

const GET_MORE_PEOPLE = "GET_MORE_PEOPLE";
const GET_MORE_PLANETS = "GET_MORE_PLANETS";
const GET_MORE_STARSHIPS = "GET_MORE_STARSHIPS";

const IS_LOADED_PEOPLE = "IS_LOADED_PEOPLE";
const IS_LOADED_PLANETS = "IS_LOADED_PLANETS";
const IS_LOADED_STARSHIPS = "IS_LOADED_STARSHIPS";

export {
	SET_FILMS_TO_STORE,
	SET_PEOPLE_TO_STORE,
	SET_PLANETS_TO_STORE,
	SET_STARSHIPS_TO_STORE,
	GET_MORE_STARSHIPS,
	GET_MORE_PEOPLE,
	GET_MORE_PLANETS,
	IS_LOADED_STARSHIPS,
	IS_LOADED_PLANETS,
	IS_LOADED_PEOPLE,
};
