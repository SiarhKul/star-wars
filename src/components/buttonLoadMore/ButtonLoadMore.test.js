import React from "react";
import { shallow } from "enzyme";
import * as redux from "react-redux";
import { ButtonLoadMore } from "..";
import { GET_MORE_PEOPLE } from "../../redux/actions/actions";

describe("Test 'ButtonLoadMore' component", () => {
	let spyOnUseDispatch;
	let mockDispatch;

	const action = {
		type: GET_MORE_PEOPLE,
	};

	beforeEach(() => {
		spyOnUseDispatch = jest.spyOn(redux, "useDispatch");
		mockDispatch = jest.fn();
		spyOnUseDispatch.mockReturnValue(mockDispatch);
	});

	afterEach(() => {
		jest.restoreAllMocks();
	});

	it("should render 'ButtonLoadMore' ", () => {
		const component = shallow(<ButtonLoadMore />);
		expect(component).toMatchSnapshot();
	});

	it(" do 'ButtonLoadMore' have a className '.btn-load-more'", () => {
		const component = shallow(<ButtonLoadMore />);
		const wrapper = component.find(".btn-load-more");
		expect(wrapper).toHaveLength(1);
	});

	it("should click on button", () => {
		const component = shallow(<ButtonLoadMore action={action} />);
		const btn = component.find(".btn-load-more");
		btn.simulate("click");
		expect(mockDispatch.mock.calls.length).toBe(1);
	});
});
