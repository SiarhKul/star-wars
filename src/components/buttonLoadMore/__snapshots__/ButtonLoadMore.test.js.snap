// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 'ButtonLoadMore' component should render 'ButtonLoadMore'  1`] = `
<div
  className="btn-load-more_center"
>
  <button
    className="btn-load-more active"
    onClick={[Function]}
  >
    Load more
  </button>
</div>
`;
